src/base/ERC721Permit.sol

```diff
@@ -2,19 +2,73 @@
 pragma solidity ^0.8.24;
 
 import {ERC721} from "solmate/src/tokens/ERC721.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC721PermitHashLibrary} from "../libraries/ERC721PermitHash.sol";
+import {SignatureVerification} from "permit2/src/libraries/SignatureVerification.sol";
 
-/// @notice An ERC721 contract that supports permit.
-/// TODO: Support permit.
-contract ERC721Permit is ERC721 {
-    constructor(string memory name_, string memory symbol_, string memory version_) ERC721(name_, symbol_) {}
+import {IERC721Permit} from "../interfaces/IERC721Permit.sol";
+import {UnorderedNonce} from "./UnorderedNonce.sol";
+
+/// @title ERC721 with permit
+/// @notice Nonfungible tokens that support an approve via signature, i.e. permit
+abstract contract ERC721Permit is ERC721, IERC721Permit, EIP712, UnorderedNonce {
+    using SignatureVerification for bytes;
+    /// @inheritdoc IERC721Permit
+    /// @dev Value is equal to keccak256("Permit(address spender,uint256 tokenId,uint256 nonce,uint256 deadline)");
+
+    bytes32 public constant override PERMIT_TYPEHASH = ERC721PermitHashLibrary.PERMIT_TYPEHASH;
```

Comment:
do we need to declare this here? don't see it used internally

Reply:
it was part of IERC721Permit, from v3

not sure if backend/interface depends on it 🤔

Reply:
i feel like its not worth an rpc call to look this up we can just calculate it offchain? i say remove